src/blockMonitor/confirmationObserver.ts

```diff
@@ -59,20 +58,21 @@ export class ConfirmationObserver extends StartStopService {
      * @param nConfirmations
      **/
     public waitForConfirmations(txHash: string, nConfirmations: number): CancellablePromise<void> {
-        if (nConfirmations > this.blockCache.maxDepth) {
+        const blockCache = this.blockProcessor.blockCache;
+        if (nConfirmations > blockCache.maxDepth) {
             return new CancellablePromise((_, reject) =>
                 reject(new ArgumentError("nConfirmations cannot be bigger than the BlockCache's maxDepth."))
             );
         }
 
         const headHash = this.blockProcessor.head && this.blockProcessor.head.hash;
-        if (headHash && this.blockCache.getConfirmations(headHash, txHash) >= nConfirmations) {
+        if (headHash && blockCache.getConfirmations(headHash, txHash) >= nConfirmations) {
```

Comment:
Is this check on head hash necessary? And the one on the line above?

Reply:
There is also a check in waitForBlocks that can be removed

Reply:
Guard removed, and it revealed a couple of tests that needed fixing due to mocking the BlockProcessor incorrectly (fixed in 82087c8d41e383cd2c3287cd87efc96da92dbb1a).